src/Plugin/VoteUpDownWidgetBase.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Drupal\vud\Plugin;
+
+use Drupal\Component\Plugin\PluginBase;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+
+/**
+ * @inheritdoc
+ */
+abstract class VoteUpDownWidgetBase extends PluginBase implements VoteUpDownWidgetInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * Get label for the plugin
+   *
+   * @return mixed
+   */
+  public function getLabel() {
+    return $this->getPluginDefinition()['label'];
+  }
+
+  /**
+   * @param $widget_template
+   * @param $variables
+   */
+  function alterTemplateVars($widget_template, &$variables) {
```

Comment:
I mentioned before:

> But more importantly, the code here should not know anything about the widget plugin instances. In other words, you should not use conditionals on the template name, but instead implement this method on each of the relevant plugins, so you use an OOP design.

That is also still pending. Do not use a conditional on the base class, instead make this method generic, and override it from other children classes when needed, that's the way OOP helps.